scripts/setup/postgres-init-db

```diff
@@ -2,32 +2,74 @@
 set -e
 set -x
 
+# A bit of a helper variable, default is voyager
+DEPLOYMENT_TYPE="${DEPLOYMENT_TYPE:-voyager}"
+# What user should we use for connecting to the database
+POSTGRES_USER="${POSTGRES_USER:-postgres}"
+
 # Shut down all services to ensure a quiescent state.
-supervisorctl stop all
+# Stop supervisord only if the socket exists, prevents problems with docker
+if [ -e "/var/run/supervisor.sock" ]; then
+    supervisorctl stop all
+fi
+
+# Don't "terminate" psql sessions when run in docker!
+if [ "$DEPLOYMENT_TYPE" != "dockervoyager" ]; then
+    # Drop any open connections to any old database.  Hackishly call using
+    # source because postgres user can't read /root/zulip/scripts/setup.
+    source "$(dirname "$0")/terminate-psql-sessions" postgres zulip zulip_base
+fi
```

Comment:
I think that the `terminate_psql_sessions` fails because it tries to connect to the localhost "only". If the script fails, the "setup" of the image fails and won't start. 

I'm going to check if I can remove the `if [ "$DEPLOYMENT_TYPE" != "dockervoyager" ]; then` here.